src/components/vuestic-components/va-input/VaInput.vue

```diff
@@ -86,120 +88,157 @@
+  },
+  placeholder: {
+    type: String,
+    default: '',
+  },
+  type: {
+    type: String,
+    default: 'text',
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+  readonly: {
+    type: Boolean,
+    default: false,
+  },
+  removable: {
+    type: Boolean,
+    default: false,
+  },
+  tabindex: {
+    type: Number,
+    default: 0,
+  },
+  errorCount: {
+    type: Number,
+    default: 1,
+  },
+  success: {
+    type: Boolean,
+    default: false,
+  },
+  messages: {
+    type: Array,
+    default () {
+      return []
     },
-    removable: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'removable', false)
-      },
+  },
+  error: {
+    type: Boolean,
+    default: false,
+  },
+  errorMessages: {
+    type: Array,
+    default () {
+      return []
     },
+  },
 
-    // textarea-specific
-    autosize: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'autosize', false)
-      },
+  // textarea-specific
+  autosize: {
+    type: Boolean,
+    default: false,
+  },
+  minRows: {
+    type: Number,
+    default: null,
+    validator: (val) => {
+      if (!(val > 0 && (val | 0) === val)) {
+        throw new Error(`\`minRows\` must be a positive integer grater than 0, but ${val} is provided`)
+      }
+      return true
     },
-    minRows: {
-      type: Number,
-      default () {
-        return getContextPropValue(this, 'minRows', null)
-      },
-      validator: (val) => {
-        if (!(val > 0 && (val | 0) === val)) {
-          throw new Error(`\`minRows\` must be a positive integer grater than 0, but ${val} is provided`)
-        }
-        return true
-      },
+  },
+  maxRows: {
+    type: Number,
+    validator: (val) => {
+      if (!(val > 0 && (val | 0) === val)) {
+        throw new Error(`\`maxRows\` must be a positive integer grater than 0, but ${val} is provided`)
+      }
+      return true
     },
-    maxRows: {
-      type: Number,
-      validator: (val) => {
-        if (!(val > 0 && (val | 0) === val)) {
-          throw new Error(`\`maxRows\` must be a positive integer grater than 0, but ${val} is provided`)
-        }
-        return true
-      },
-      default () {
-        return getContextPropValue(this, 'maxRows', null)
-      },
+    default: null,
+  },
+  rules: {
+    type: Array,
+    default () {
+      return []
     },
   },
+})
+
+const prepareValidations = (messages = [], callArguments = null) =>
+  messages
+    .map((message) => isFunction(message) ? message(callArguments) : message)
+    .filter(Boolean)
+
+export default {
+  name: 'VaInput',
+  mixins: [ColorThemeMixin, InputContextMixin],
+  components: { VaInputWrapper, VaIcon },
   mounted () {
     this.adjustHeight()
   },
   watch: {
     value () {
-      this.adjustHeight()
+      if (this.isTouchedValidation) {
+        this.validate()
+      }
+    },
+    error: {
+      handler (error) {
+        this.internalError = error
```

Comment:
We cannot use computed, because we need to set internal error prop on validating field with rules.
I don't see a better way to sets an internal error on change error props with bindings and handle rule validation errors together.

Reply:
I have no idea how to make reset-validation method with computed props

Reply:
internal prop? o.o
Internal state should not change because of error prop update. Please use computed.